library/ssl_msg.c

```diff
@@ -574,6 +578,28 @@ int mbedtls_ssl_encrypt_buf( mbedtls_ssl_context *ssl,
         return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
     }
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+    if( transform->minor_ver == MBEDTLS_SSL_MINOR_VERSION_4 )
+    {
+        /*
+         * Wrap plaintext into TLSInnerPlaintext structure.
+         * See ssl_build_inner_plaintext() for more information.
+         *
+         * Note that this changes `rec->data_len`, and hence
+         * `post_avail` needs to be recalculated afterwards.
+         */
+        if( ssl_build_inner_plaintext( data,
```

Comment:
Suggestion: We're calling `ssl_build_inner_plaintext()` here and below. Is it possible for `transform->minor_ver == MBEDTLS_SSL_MINOR_VERSION_4` and `transform->out_cid_len != 0`, in which case would it be better and simpler to combine the two potential calls to `ssl_build_inner_plaintext()`?

Reply:
Good point, I wondered about that, too. For now I have deliberately kept the code paths separate because there's no overlap of the TLS 1.3 code and CID code. 

That said, at least a comment saying as much might be helpful.

Reply:
Done.